app/src/main/java/de/tum/in/tumcampusapp/component/ui/tufilm/show_ticket.java

```diff
@@ -14,30 +15,31 @@
 import com.journeyapps.barcodescanner.BarcodeEncoder;
 
 import de.tum.in.tumcampusapp.R;
+import de.tum.in.tumcampusapp.component.other.generic.activity.BaseActivity;
+import de.tum.in.tumcampusapp.utils.Const;
 
-public class show_ticket extends AppCompatActivity {
+public class show_ticket extends BaseActivity {
 
     private TextView moviedetailTextView;
     private TextView ticketsnumberTextView;
     private ImageView ticketqrcode;
-    /*
-        public TuitionFeesActivity() {
-                super(TUMOnlineConst.Companion.getTUITION_FEE_STATUS(), R.layout.activity_tuitionfees);
-            }
-    */
+
+    public show_ticket() {
+        super(R.layout.activity_show_ticket);
+    }
     @Override
     public void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
-
-        moviedetailTextView = findViewById(R.id.moviedetail);
-        ticketsnumberTextView = findViewById(R.id.ticketnumber);
-        ticketqrcode = findViewById(R.id.ticket_qrcode);
-        moviedetailTextView.setText(" KingsMan 08.05 16:00-17:00\n" +
-                "1. Stock, Hörsaal 1200 (Carl-von-Linde-Hörsaal)\n" +
-                "Arcisstraße 21");//data from last activity
+        moviedetailTextView = (TextView)findViewById(R.id.moviedetail);
+        ticketsnumberTextView = (TextView)findViewById(R.id.ticketnumber);
+        ticketqrcode = (ImageView) findViewById(R.id.ticket_qrcode);
+        //get data from last activity
+        Intent intent = getIntent();
+        String data = getIntent().getStringExtra("movie_data");
+        moviedetailTextView.setText(data);
         ticketsnumberTextView.setText("87237489273984");//data from backend
-        //ticketqrcode.setImageResource();
-        String text="KingsMan 08.05 Kailiang Dong 03695880"; // Whatever you need to encode in the QR code
+
+        String text="KingsMan 08.05 Kailiang Dong "; // Whatever you need to encode in the QR code
         MultiFormatWriter multiFormatWriter = new MultiFormatWriter();
```

Comment:
Maybe outsource the creation of the barcode to a seperate function for better code structure.

Reply:
I used a libraray of QR code(zxing). And waiting for string from backend(last time we said we would discuss what kind of string we will make last time.)

For outsource the creation of the barcode , do you mean make a method “ public void QrcodeCreation( String text){} outside this class?

Reply:
Not **outside** the class, but rather a "private void createQRCode(String text) {...}" inside the class. I don't like putting everything in onCreate.

Reply:
Yes, absolutely a great idea to create smaller functions that do one thing. Creating a function for QR Code generation is a good idea!